lib/os/mempool.c

```diff
@@ -56,6 +56,14 @@ static void clear_alloc_bit(struct sys_mem_pool_base *p, int level, int bn)
 	*word &= ~(1<<bit);
 }
 
+static inline bool alloc_bit_is_set(struct sys_mem_pool_base *p, int level, int bn)
+{
+	u32_t *word;
+	int bit = get_bit_ptr(p, level, bn, &word);
+
+	return !!(*word & (1<<bit));
```

Comment:
`(1 << bit)` should be spelled `BIT(bit)` using `sys/util.h` which correctly handles the case of the high bit of an unsigned type.  (`1 << 31` is undefined for signed 32-bit systems.)

It appears there are many other occurrences of `(1 << bit)` in this file that should also be fixed.

If MISRA compliance is important the return value should be something like `(*word & BIT(bit)) != 0U)`.